test/adapter/json_api/resource_type_config_test.rb

```diff
@@ -52,6 +57,13 @@ def test_config_singular
               end
             end
           end
+
+          def test_explicit_type_value
+            with_adapter :json_api do
+              hash = ActiveModel::SerializableResource.new(@author, serializer: ProfileTypeSerializer).serializable_hash
+              assert_equal('profile', hash[:data][:type])
```

Comment:
`with_adapter` is a [global thing that needs to go away](https://github.com/rails-api/active_model_serializers/blob/master/test/support/serialization_testing.rb#L8-L12).  if you don't mind, let's

```ruby
serialized_resource = ActiveModel::SerializableResource.new(@author, serializer: ProfileTypeSerializer, adapter: :json_api).serializable_hash
assert_equal('profile', serialized_resource.fetch(:data).fetch(:type))
```

(the fetch is just to get a better exception if serialized_resource isn't expected)

if you want to [take advantage of a future test simplification](https://github.com/rails-api/active_model_serializers/pull/1050/files#diff-02cd2f48e8583af0b097a44913eb31baR33), add the private method `serializable` to this test

```ruby
def serializable(resource, options = {})
    ActiveModel::SerializableResource.new(resource, options)
  end
```

so then you have `hash = serializable(@author, serializer: ..., adapter: :json_api).serializable_hash`